backend/src/main/java/com/huellapositiva/application/controller/ProposalApiController.java

```diff
@@ -544,5 +553,52 @@ public void changeStatusVolunteerInProposal(@RequestBody List<ChangeStatusVolunt
     public void changeReviewPendingProposalToPublished(@RequestBody ChangeStatusProposalRequestDto dto) {
         changeReviewPendingProposalToPublishedAction.execute(dto.getIdProposal());
     }
-}
 
+    @Operation(
+            summary = "Updates the proposal image",
+            description = "The contact person can update the image of the proposal.",
+            tags = {"proposals, contact person"},
+            parameters = {
+                    @Parameter(name = "X-XSRF-TOKEN", in = ParameterIn.HEADER, required = true, example = "ff79038b-3fec-41f0-bab8-6e0d11db986e", description = "For taking this value, open your inspector code on your browser, and take the value of the cookie with the name 'XSRF-TOKEN'. Example: a6f5086d-af6b-464f-988b-7a604e46062b"),
+                    @Parameter(name = "XSRF-TOKEN", in = ParameterIn.COOKIE, required = true, example = "ff79038b-3fec-41f0-bab8-6e0d11db986e", description = "Same value of X-XSRF-TOKEN")
+            },
+            security = {
+                    @SecurityRequirement(name = "accessToken")
+            }
+    )
+    @ApiResponses(
+            value = {
+                    @ApiResponse(
+                            responseCode = "204",
+                            description = "No Content, proposal status changed to PUBLISHED successfully."
+                    ),
+                    @ApiResponse(
+                            responseCode = "400",
+                            description = "Bad Request, The proposal status or image not valid"
+                    ),
+                    @ApiResponse(
+                            responseCode = "403",
+                            description = "Forbidden, The contact person related to this proposal does not match the logged contact person.."
+                    ),
+                    @ApiResponse(
+                            responseCode = "500",
+                            description = "Internal server error, could not fetch the user data due to a connectivity issue."
+                    )
+            }
+    )
+    @PostMapping("/udpateProposalImage")
```

Comment:
PUT /{id}/image -> RESTful URLs

where {id} would be the proposalId

Reply:
Solved